first review

George051191 opened the pull request.

marias1403 committed:
feat: QuizCardList

marias1403 committed:
Merge pull request #26 from Corporate-Quiz/feat/quiz-card

feat: QuizCardList

marias1403 committed:
fix: QuizCard, QuizCardList

marias1403 committed:
Merge pull request #27 from Corporate-Quiz/feat/quiz-card

fix: QuizCard, QuizCardList

marias1403 committed:
fix: letter spacing

marias1403 committed:
Merge pull request #28 from Corporate-Quiz/feat/quiz-card

fix: letter spacing

FrantsuzovaTamara committed:
change quiz menu styles

FrantsuzovaTamara committed:
chenge structure of image files

marias1403 committed:
fix: QuizCard tags

marias1403 committed:
Merge pull request #29 from Corporate-Quiz/feat/quiz-card

fix: QuizCard tags

FrantsuzovaTamara committed:
add logic in castle section

George051191 committed:
fixes done

George051191 committed:
Merge pull request #30 from Corporate-Quiz/hotfix/allpages

fixes done

George051191 committed:
fixes

marias1403 committed:
feat: QuizCard animation added

marias1403 committed:
Merge pull request #31 from Corporate-Quiz/feat/quiz-card

feat: QuizCard animation added

marias1403 committed:
fix: QuizCard Animation

marias1403 committed:
Merge pull request #32 from Corporate-Quiz/feat/quiz-card

fix: QuizCard Animation

marias1403 committed:
Merge pull request #32 from Corporate-Quiz/feat/quiz-card

fix: QuizCard Animation

FrantsuzovaTamara committed:
add logic for quiz-filter

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

George051191 committed:
add methods for data

George051191 committed:
Merge branch 'develop' into fix/rtkfetch

FrantsuzovaTamara committed:
fix filter's logic after chenging the path

FrantsuzovaTamara committed:
Merge branch 'quiz-filter' into develop

George051191 committed:
lazy add

MaxOrel commented on src/ui-lib/widgets/QuizCardList.tsx:
```diff
@@ -0,0 +1,60 @@
+}
+
+const QuizCardList: FC = () => {
+  const { quizzesOnPage } = useSelector((state) => state.all);
```
А одной деструктуризацией не обойдемся?

George051191 committed:
Merge branch 'develop' into fix/rtkfetch

George051191 committed:
fixes

MaxOrel commented on src/store/allSlice.tsx:
```diff
@@ -0,0 +1,70 @@
+};
+
+const allSlice = createSlice({
+  name: 'allData',
```
Хорошей практикой является вынесение имени среза в константу, чтобы можно было его пере использовать. Например создайте папку для каждого среза, помимо файла среза положите рядом файл файл selector и в нем опишите селекторы которыми будете пользоваться, так вы улучшите поддержку проекта при изменении структуры стора.
Вот пример:  
![image](https://github.com/Corporate-Quiz/frontend/assets/27287886/99d32675-38ab-4030-853d-53ba8a1d6833)
![image](https://github.com/Corporate-Quiz/frontend/assets/27287886/23d2e373-c621-41f9-9631-7391d843ed2c)
![image](https://github.com/Corporate-Quiz/frontend/assets/27287886/fb49343c-c283-4150-b9ce-86c88587de73)

George051191 committed:
Merge pull request #33 from Corporate-Quiz/fix/rtkfetch

Fix/rtkfetch

MaxOrel commented on src/pages/Login.tsx:
```diff
@@ -0,0 +1,267 @@
+  }
+`;
+
+const Login: React.FC = () => {
```
Я понимаю что у вас нет адаптива, но десктопные устройства тоже разные бывают , кажется что отдавать приоритет блоку слева не лучшая идея 
![image](https://github.com/Corporate-Quiz/frontend/assets/27287886/19b4d8fa-ba4d-4a76-b290-4f1d139aa9bd)

MaxOrel commented on src/ui-lib/widgets/Header.tsx:
```diff
@@ -0,0 +1,97 @@
+  return (
+    <HeaderWrapper>
+      <HeaderContainer>
+        <AdvBanner isOpen={isOpen} />
```
Реализуйте закрытие при нажатии вне области модалки с уведомлениями.

MaxOrel commented:
В целом мне понравилось как вы организовали структуру проекта, используете RTK и планируете использовать RTQ. Общий коммент по верстке: даже если мы делаем верстку для стационарной версии, то в голове всегда держим что будет адаптив и по стараемся не допускать ошибок которые приведут к создании кучи брекпоинтов и переписывании стационарной версии.

George051191 committed:
fixes

George051191 committed:
Merge pull request #34 from Corporate-Quiz/fix/userlk

fixes

George051191 committed:
list added

George051191 committed:
Merge pull request #35 from Corporate-Quiz/feat/listedu

list added

George051191 committed:
fixes done

George051191 committed:
Merge pull request #36 from Corporate-Quiz/hot-fix/review1

fixes done

FrantsuzovaTamara committed:
create page with information about quiz

George051191 committed:
fix query

George051191 committed:
Merge pull request #37 from Corporate-Quiz/fix/onequiz

fix query

FrantsuzovaTamara committed:
fix castle's button func and chenge filter in quizzes

George051191 committed:
bar done

George051191 committed:
Merge branch 'develop' into feat/quizsidebar

George051191 committed:
Merge pull request #38 from Corporate-Quiz/feat/quizsidebar

Feat/quizsidebar

George051191 committed:
fix url

marias1403 committed:
feat: QuizQuestion

marias1403 committed:
Merge branch 'develop' into feat/quiz-question

# Conflicts:
#	src/constants/routes.tsx

marias1403 committed:
feat: QuizQuestion layout

marias1403 committed:
Merge pull request #39 from Corporate-Quiz/feat/quiz-question

Feat/quiz question

FrantsuzovaTamara committed:
create loader

FrantsuzovaTamara committed:
Merge branch 'loader' into develop

George051191 committed:
fix token

George051191 committed:
Merge pull request #40 from Corporate-Quiz/test

fix token

George051191 committed:
Add files via upload

George051191 committed:
fixes done

George051191 committed:
Merge branch 'develop' into feat/alldone

George051191 committed:
fixes

George051191 committed:
Merge pull request #41 from Corporate-Quiz/feat/alldone

Feat/alldone

George051191 committed:
fixes

George051191 committed:
build done

FrantsuzovaTamara committed:
fix styles and logic

George051191 committed:
fixes

George051191 committed:
fix done

FrantsuzovaTamara committed:
change rating photos

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
change styles in adv banner and fix filter in quizzes

FrantsuzovaTamara committed:
add logic and styles for selected answers

FrantsuzovaTamara committed:
fix text

George051191 committed:
some hot fixes

marias1403 committed:
feat: QuizErrorsReview

FrantsuzovaTamara committed:
fix styles

marias1403 committed:
feat: QuizErrorsReview expanded list added

marias1403 committed:
feat: QuizErrorsReview check icon added

marias1403 committed:
Merge pull request #42 from Corporate-Quiz/feat/quiz-error-review

Feat/quiz error review

FrantsuzovaTamara committed:
fix styles

marias1403 committed:
fix: layout

marias1403 committed:
Merge pull request #43 from Corporate-Quiz/fix/pp-layout-password-review

fix: layout

George051191 committed:
fixes done

George051191 committed:
Merge pull request #44 from Corporate-Quiz/hot-fix/bags

fixes done

bigfuto committed:
Create workflow.yml

marias1403 committed:
feat: QuizErrorsReview stata

marias1403 committed:
Merge pull request #45 from Corporate-Quiz/feat/quiz-error-review

feat: QuizErrorsReview stata

marias1403 committed:
fix: Dropdown and QuizErrorsReview

marias1403 committed:
Merge pull request #46 from Corporate-Quiz/feat/quiz-error-review

fix: Dropdown and QuizErrorsReview

FrantsuzovaTamara committed:
bugs fix

George051191 committed:
fixes done

George051191 committed:
Merge branch 'develop' into fix/quizgo

George051191 committed:
fixes done

George051191 committed:
Merge pull request #47 from Corporate-Quiz/fix/quizgo

Fix/quizgo

FrantsuzovaTamara committed:
fix styles

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
add right answers counter

marias1403 committed:
fix: QuizErrorsReview

marias1403 committed:
Merge pull request #48 from Corporate-Quiz/feat/quiz-error-review

fix: QuizErrorsReview

marias1403 committed:
fix: QuizErrorsReview

marias1403 committed:
Merge pull request #49 from Corporate-Quiz/feat/quiz-error-review

fix: QuizErrorsReview

marias1403 committed:
fix: date

marias1403 committed:
Merge pull request #50 from Corporate-Quiz/feat/quiz-error-review

fix: date

FrantsuzovaTamara committed:
fix loader and castle's link

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

marias1403 committed:
fix: QuizErrorsReview

marias1403 committed:
Merge pull request #51 from Corporate-Quiz/feat/quiz-error-review

fix: QuizErrorsReview

FrantsuzovaTamara committed:
add error parsing on page with quiz

George051191 committed:
fixes

FrantsuzovaTamara committed:
fix bug with cash

marias1403 committed:
feat: SingleChoiceQuestion, MultipleChoiceQuestion, OpenEndedQuestion, DragAndDropQuestion

marias1403 committed:
Merge pull request #52 from Corporate-Quiz/feat/questions-types

feat: SingleChoiceQuestion, MultipleChoiceQuestion, OpenEndedQuestion…

marias1403 committed:
feat: MultipleChoiceAnswer

marias1403 committed:
Merge pull request #53 from Corporate-Quiz/feat/questions-types

feat: MultipleChoiceAnswer

marias1403 committed:
fix: QuizQuestion

marias1403 committed:
Merge pull request #54 from Corporate-Quiz/feat/questions-types

fix: QuizQuestion

FrantsuzovaTamara committed:
add admin link

marias1403 committed:
feat: Achievements

marias1403 committed:
Merge pull request #55 from Corporate-Quiz/feat/achievements

feat: Achievements

marias1403 committed:
feat: AchievementPage added

marias1403 committed:
Merge pull request #56 from Corporate-Quiz/feat/achievements

feat: AchievementPage added

marias1403 committed:
AchievementPage added

marias1403 committed:
Merge pull request #57 from Corporate-Quiz/feat/achievements

AchievementPage added

FrantsuzovaTamara committed:
feat: верстка нового экрана + защита роутов в зависимости от роли

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'staff' into develop

marias1403 committed:
feat: Learning Progress

marias1403 committed:
Merge branch 'develop' into feat/quiz-error-review

marias1403 committed:
fix: PersonalAccount

marias1403 committed:
Merge pull request #58 from Corporate-Quiz/feat/quiz-error-review

Feat/quiz error review

FrantsuzovaTamara committed:
feat: add scroll on table, buttons for scroll and checkbox group

FrantsuzovaTamara committed:
Merge branch 'staff' into develop

marias1403 committed:
feat: IncompleteQuizzes

marias1403 committed:
Merge branch 'develop' into feat/incomplete-quizzes

marias1403 committed:
Merge pull request #59 from Corporate-Quiz/feat/incomplete-quizzes

Feat/incomplete quizzes

FrantsuzovaTamara committed:
add functions for popups

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

marias1403 committed:
feat: AvatarUploadPopup

marias1403 committed:
Merge branch 'develop' into feat/avatar-upload-popup

# Conflicts:
#	src/ui-lib/widgets/Header.tsx

marias1403 committed:
Merge pull request #61 from Corporate-Quiz/feat/avatar-upload-popup

Feat/avatar upload popup

FrantsuzovaTamara committed:
fix: fix 404 errors on routes

FrantsuzovaTamara committed:
feat: add admin api

FrantsuzovaTamara committed:
feat: add new-quiz page

FrantsuzovaTamara committed:
feat: add step1 on quiz creation + make a progress bar

FrantsuzovaTamara committed:
feat: add step2 page, types for questions

juliv committed:
feat: #60 Стилизовать страницу Error

juliv committed:
Merge branch 'develop' into feat/60-page-404

juliv committed:
refactor: #62 remove directory `/build`

juliv committed:
refactor: #62 rename tsx->ts

juliv committed:
feat: #62 update ts-types for images

juliv committed:
refactor: #62 refactor eslint, tsconfig, gitignore; add editorconfig

juliv committed:
refactor: #62 move images to assets, update import paths (started with @/)

juliv committed:
ci: #62 add `@babel/plugin-proposal-private-property-in-object`

juliv committed:
fix: #60 update import's paths

juliv committed:
Merge pull request #63 from Corporate-Quiz/feat/60-page-404

feat: Page 404

juliv committed:
Merge branch 'develop' into refactor/62-refactor

juliv committed:
refactor: #62 update Page404

juliv committed:
Merge pull request #64 from Corporate-Quiz/refactor/62-refactor

refactor: #62 Рефакторинг проекта

FrantsuzovaTamara committed:
feat: adding and delete  questions and answers on page

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
fix: types

juliv committed:
ci: #66 update package.json

juliv committed:
refactor: #66 update eslintrc, editorconfig

juliv committed:
refactor: #66 add types, interfaces

juliv committed:
Merge pull request #67 from Corporate-Quiz/refactor/66-eslint

refactor: #66 eslint

juliv committed:
refactor: #66 update css

juliv committed:
ci: #66 update packages

juliv committed:
ci: #66 update docker

juliv committed:
Merge pull request #68 from Corporate-Quiz/refactor/66-eslint

refactor: #66 eslint

FrantsuzovaTamara committed:
feat: add drag and drop question with several boards

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
fix: fix eslint errors

juliv committed:
fix: #65 fix login error

juliv committed:
ci: update workflow

juliv committed:
fix: disallow eslint-rule `operator-linebreak`

juliv committed:
Merge pull request #69 from Corporate-Quiz/fix/fix-login-error

fix: fix login error

juliv committed:
ci: update job versions in workflow

juliv committed:
refactor: optimize images

juliv committed:
Merge branch 'develop' into feat/65-new-quiz-step3

juliv committed:
fix: fix show login form

juliv committed:
Merge branch 'develop' into feat/65-new-quiz-step3

FrantsuzovaTamara committed:
fix: fixeslint  errors

juliv committed:
feat: add plural forms

marias1403 committed:
feat: Achievements backend

juliv committed:
Merge branch 'develop' into feat/65-new-quiz-step3

juliv committed:
refactor: typing and refactor API-requests

marias1403 committed:
feat: Rating, Achievements, Progress backend

marias1403 committed:
Merge pull request #70 from Corporate-Quiz/feat/user-main-page

feat: Rating, Achievements, Progress backend

juliv committed:
Merge branch 'develop' into feat/65-new-quiz-step3

# Conflicts:
#	src/ui-lib/widgets/LearningProgress.tsx

juliv committed:
refactor: refactor LearningProgress

juliv committed:
feat: add draft NewQuiz (step 3)

juliv committed:
Merge pull request #71 from Corporate-Quiz/feat/65-new-quiz-step3

feat: #65 new quiz step3

marias1403 committed:
fix: AvatarPopup

marias1403 committed:
Merge branch 'develop' into feat/user-main-page

marias1403 committed:
Merge pull request #72 from Corporate-Quiz/feat/user-main-page

Feat/user main page

marias1403 committed:
fix: Castle component backend

marias1403 committed:
Merge pull request #75 from Corporate-Quiz/feat/user-main-page

fix: Castle component backend

marias1403 committed:
feat: AvatarUpload backend

marias1403 committed:
Merge pull request #76 from Corporate-Quiz/feat/user-main-page

feat: AvatarUpload backend

FrantsuzovaTamara committed:
fix: logic on second step

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'new-quiz-step2' into develop

FrantsuzovaTamara committed:
feat: add functions for drag and drop question

marias1403 committed:
feat: MultipleChoiceQuestion, SingleChoiceQuestion, OpenEndedQuestion backend

FrantsuzovaTamara committed:
feat: start making connection with server

marias1403 committed:
feat: DragAndDropQuestion backend

marias1403 committed:
Merge pull request #77 from Corporate-Quiz/feat/user-main-page

Feat/user main page

juliv committed:
feat: add RTK-query for admin_* sections

- admin_tags
- admin_levels
- admin_departments
- admin_quizes
- admin_questions
- admin_volumes
- admin_users

juliv committed:
Merge pull request #78 from Corporate-Quiz/feat/74-add-rtk-queries

feat: #74 add RTK-query for admin_* sections

FrantsuzovaTamara committed:
feat: add step4, gallery popup and start preview popup

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

juliv committed:
refactor: #79 use constants in urls and pronouns

juliv committed:
Merge pull request #80 from Corporate-Quiz/refactor/79-refactor-lk

refactor: #79 use constants in urls and pronouns

FrantsuzovaTamara committed:
feat: add connection wir=th server for staff page

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

marias1403 committed:
feat: IncompleteQuizzes

marias1403 committed:
Merge branch 'develop' into feat/user-main-page

marias1403 committed:
feat: Incomplete Quizzes

marias1403 committed:
Merge pull request #81 from Corporate-Quiz/feat/user-main-page

Feat/user main page

marias1403 committed:
fix: incomplete quizzes

marias1403 committed:
Merge pull request #82 from Corporate-Quiz/feat/user-main-page

fix: incomplete quizzes

marias1403 committed:
fix: answers review

marias1403 committed:
fix: quiz answer review

marias1403 committed:
Merge pull request #83 from Corporate-Quiz/feat/user-main-page

Feat/user main page

FrantsuzovaTamara committed:
add connection with server for 1 and 4 quiz steps + make fixes

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

marias1403 committed:
feat: DropdownOneAnswersList, DropdownMnyAnswersList, DropdownOpnAnswersList

marias1403 committed:
Merge branch 'develop' into feat/user-main-page

marias1403 committed:
Merge pull request #84 from Corporate-Quiz/feat/user-main-page

Feat/user main page

marias1403 committed:
fix: IncompleteQuizzes

marias1403 committed:
Merge pull request #85 from Corporate-Quiz/feat/user-main-page

fix: IncompleteQuizzes

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

marias1403 committed:
feat: DropdownLstAnswersList

marias1403 committed:
Merge pull request #86 from Corporate-Quiz/feat/user-main-page

feat: DropdownLstAnswersList

juliv committed:
feat: add step 3 in create quiz

juliv committed:
Merge branch 'develop' into feat/new-quiz-step3

# Conflicts:
#	src/ui-lib/widgets/NewQuizStep3.tsx

juliv committed:
fix: fix 3 step create quiz after merge

FrantsuzovaTamara committed:
add connection with server for second step + fix popup

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

juliv committed:
feat: autosave step 3

FrantsuzovaTamara committed:
add fixes

FrantsuzovaTamara committed:
fixes

juliv committed:
Merge branch 'develop' into feat/new-quiz-step3

juliv committed:
Merge pull request #87 from Corporate-Quiz/feat/new-quiz-step3

feat: new quiz step3

FrantsuzovaTamara committed:
fixes

FrantsuzovaTamara committed:
Merge branch 'develop' of https://github.com/Corporate-Quiz/frontend into develop

juliv committed:
refactor: #88 correct eslint rules, remove unnecessary `eslint-disable` sections

juliv committed:
refactor: #88 correct css styles

juliv committed:
refactor: #88 correct variable names

juliv committed:
refactor: #88 rename tsx->ts (api, store)

juliv committed:
refactor: #88 update api rout's names

juliv committed:
Merge pull request #89 from Corporate-Quiz/feat/88-eslint-correct

Feat/88 eslint correct
